### DISCUSSION TITLE
rp6502 running 65C816

### DISCUSSION THREAD

**Discussion by basic-xl** [picocomputer/community#93] [Q&A]
Hello! I am Jesse from Oklahoma. My first post here on the rp6502 forums. Some time back, I had asked @rumbledethumps about the possibility of creating a 16mb 65c816 version as well, and unfortunately for me, he is way too busy with this project. So I wondered if an 816 would work in emulated mode in the rp6502. And I am happy to report that it does! Even at 8mhz. But not 14mhz because the OS throttles PHI2 to 8mhz. Perhaps a future OS update could 'sense' an 816 and allow it to run at 14mhz? hint hint 😁
![20240302_091546](https://github.com/picocomputer/community/assets/112533139/1a3f685a-d4bf-4ac8-aa2f-0bc10651dcd4)

Keep in mind it will still be limited to 64k ram, and will have to stay in emulated mode, but if I'm not mistaken, you'll have access to 16 bit registers and branches, the built in blitter instructions, and lots more. If there's any interest, someone could comment with a concrete list of additional features.

If you'd like to order one, here's the Mouser link:
https://www.mouser.com/ProductDetail/Western-Design-Center-WDC/W65C816S6PG-14?qs=opBjA1TV900oFHK2UUDZKA%3D%3D

**Comment by basic-xl** [comment]
I just updated to 0.7 firmware and installed my first rom, Paint. But it wouldn't run. It errored out with the message "?watchdog: timeout". I figured there'd be a 'gotcha' in there somewhere, as the 816 isn't a true 100% compatible swap for the 6502. So back in goes the 6502 😉

If anyone is having trouble installing their first rom, as I did, here's the short & sweet instructions:

Put the .rp6502 rom file on the sdcard, NOT to the RIA file system as the firmware files get installed.
Put the sdcard back in the RP6502 computer and turn it on.
Type the following (example is paint.rp6502):
install paint.rp6502
NOT: install paint
NOT: install "paint"
NOT: install "paint.rp6502"
NOT: install "1:paint.rp6502"
NOT: install 1:paint.rp6502

of course I had to try everything (almost a full hour wasted lol) before I stumbled on the correct answer. >whew!<

**Comment by basic-xl** [comment]
@rumbledethumps after upgrading from 0.2 to 0.7, I had the littlefs warnings from both the ria and vga. All I could find was a vague mention to "change a setting to automatically update the filesystem", but the exact setting and where to find it, weren't mentioned. I noticed after rebooting several times trying to install the paint rom file that the error messages just disappeared, and now it boots cleanly. I have no idea what I did to make it go away.

**Reply by rumbledethumps** [reply]
The message is harmless and goes away with any write to the RIA flash.
